src/xenia/kernel/XLiveAPI.cpp

```diff
@@ -127,6 +129,51 @@ std::vector<std::string> XLiveAPI::ParseAPIList() {
   return api_addresses;
 }
 
+std::vector<std::string> XLiveAPI::ParseDelimitedList(std::string list,
+                                                      uint32_t count) {
+  std::unordered_set<std::string> unique_token;
+  std::vector<std::string> tokens;
+
+  std::stringstream token_stream(list);
+  std::string token;
+
+  while (std::getline(token_stream, token, ',')) {
+    if (tokens.size() >= count) {
+      break;
+    }
+
+    token = xe::string_util::trim(token);
```

Comment:
It will be cheaper to convert them to uint64_t already and pass as it instead of creating string for each one of them